drupal/sites/default/boinc/modules/boincwork/includes/boincwork.helpers.inc

```diff
@@ -531,33 +555,58 @@ function boincwork_generate_prefs_element(&$form, $type, $elements, $user_prefs
     }
     break;
     
-  case 'apps':
+  case 'app_overrides':
     $title = is_array($elements['title']) ? $elements['title']['@value'] : $elements['title'];
-    
-      // Translate elements as appropriate
-      if ($title) {
-        i18nstrings_update('project:prefs_xml', $title);
-        $title = i18nstrings('project:prefs_xml', $title);
-      }
-    
+
+    // Translate elements as appropriate
+    if ($title) {
+      i18nstrings_update('project:prefs_xml', $title);
+      $title = i18nstrings('project:prefs_xml', $title);
+    }
+
+    // Create the application section of the form
     $form['applications'] = array(
       '#title' => bts('Applications', array(), NULL, 'boinc:account-preferences'),
       '#type' => 'fieldset',
       '#description' => $title,
       '#collapsible' => TRUE,
       '#collapsed' => FALSE
     );
-    $applications = array();
-    if (!is_array($user_prefs['app_id'])) {
-      $user_prefs['app_id'] = array($user_prefs['app_id']);
+
+    // Two different XML formats are supported. The newer format has
+    // an <apps_selected> container with <app_id> elements as
+    // children. The older format has <app_id> elements only.
+    if (is_array($user_prefs['apps_selected']['app_id'])) {
+      $myapp_ids = $user_prefs['apps_selected']['app_id'];
+    }
+    else if (is_array($user_prefs['app_id'])) {
+      $myapp_ids = $user_prefs['app_id'];
+    }
+    // Fix the array in the case where there is exactly one
+    // app_id. myapp_ids should be an associative array where the
+    // values are themselves an array. In effect the app_id array
+    // needs to be pushed down onre more level in the tree.
+    if (is_array($myapp_ids) AND !is_array($myapp_ids[1])) {
```

Comment:
$myapp_ids is not defined if neither of the above `if ... elseif`  conditions are met. This should be initialized to NULL at least to limit the possibility of bugs / unintended behavior popping up later.